lib/apiservers/engine/backends/container.go

```diff
@@ -217,38 +217,103 @@ func (c *Container) ContainerCreate(config types.ContainerCreateConfig) (types.C
 	// commit the create op
 	_, err = client.Containers.Commit(containers.NewCommitParams().WithHandle(h))
 	if err != nil {
-		return types.ContainerCreateResponse{}, derr.NewErrorWithStatusCode(err, http.StatusInternalServerError)
+		// FIXME: Containers.Commit returns more errors than it's swagger spec says.
+		// When no image exist, it also sends back non swagger errors.  We should fix
+		// this once Commit returns correct error codes.
+		return types.ContainerCreateResponse{}, derr.NewRequestNotFoundError(fmt.Errorf("No such image: %s", layer.ID))
 	}
 
+	// Container created ok, overwrite the container params in the container store as
+	// these are the parameters that the containers were actually created with
+	layer.Config.Cmd = config.Config.Cmd
+	layer.Config.WorkingDir = config.Config.WorkingDir
+	layer.Config.Entrypoint = config.Config.Entrypoint
+	layer.Config.Env = config.Config.Env
+	layer.Config.AttachStdin = config.Config.AttachStdin
+	layer.Config.AttachStdout = config.Config.AttachStdout
+	layer.Config.AttachStderr = config.Config.AttachStderr
+	layer.Config.Tty = config.Config.Tty
+	layer.Config.OpenStdin = config.Config.OpenStdin
+	layer.Config.StdinOnce = config.Config.StdinOnce
+	layer.ContainerID = createResults.Payload.ID
+
+	viccontainer.GetCache().SaveContainer(createResults.Payload.ID, layer)
```

Comment:
Never mind this as I just saw that SaveContainer uses a mutex- see this wouldn't happen if we were using gerrit :)